specs/Cryptography/Ed25519.md

```diff
@@ -0,0 +1,69 @@
+# Ed25519
+
+Reference: https://tools.ietf.org/html/rfc8032#section-5.1
+
+## Public key
+```rust
+pub struct ED25519PublicKey(pub [u8; ed25519_dalek::PUBLIC_KEY_LENGTH]);
+```
+
+`ED25519PublicKey` is 32 bytes in size and represents a curve point in `CompressedEdwardsY` format:
+```rust
+/// In "Edwards y" / "Ed25519" format, the curve point \\((x,y)\\) is
+/// determined by the \\(y\\)-coordinate and the sign of \\(x\\).
+///
+/// The first 255 bits of a `CompressedEdwardsY` represent the
+/// \\(y\\)-coordinate.  The high bit of the 32nd byte gives the sign of \\(x\\).
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct CompressedEdwardsY(pub [u8; 32]);
+```
+
+## Secret key
+
+`ED25519SecretKey` is a 64 byte array representing a serialized keypair
+```rust
+// This is actually a keypair, because ed25519_dalek api only has keypair.sign
+// From ed25519_dalek doc: The first SECRET_KEY_LENGTH of bytes is the SecretKey
+// The last PUBLIC_KEY_LENGTH of bytes is the public key, in total it's KEYPAIR_LENGTH
+pub struct ED25519SecretKey(pub [u8; ed25519_dalek::KEYPAIR_LENGTH]);
```

Comment:
Come on, this should be a specification of cryptography used in Near, not an API reference of ed25519-dalek crate.